cse-spring-boot-starter-discovery/src/main/java/com/huawei/cse/springboot/starter/discovery/CseDiscoveryClient.java

```diff
@@ -42,45 +51,25 @@ public String description() {
         ServiceRegistryClient client = RegistryClientFactory.getRegistryClient();
         String appId = DynamicPropertyFactory.getInstance().getStringProperty("APPLICATION_ID", "default").get();
-        String versionRule = DynamicPropertyFactory.getInstance().getStringProperty("service_description.version", "1.0.0").get();
-        String cseServiceID = client.getMicroserviceId(appId , serviceId, versionRule);
+        ReferenceConfig referenceConfig = consumerProviderManager.getReferenceConfig(serviceId);
+        String versionRule = referenceConfig.getMicroserviceVersionRule();
+        String cseServiceID = client.getMicroserviceId(appId, serviceId, versionRule);
         List<MicroserviceInstance> cseServices = client.getMicroserviceInstance(cseServiceID, cseServiceID);
-        if(null != cseServices && !cseServices.isEmpty())
-        {
-        for(MicroserviceInstance instance:cseServices) 
-        {
-            List<String> eps=instance.getEndpoints();
-            for(String ep:eps)
-            {
-                URIEndpointObject uri = new URIEndpointObject(ep);
-                instances.add(new DefaultServiceInstance(instance.getServiceId(), uri.getHostOrIp(), uri.getPort(), false));
+        if (null != cseServices && !cseServices.isEmpty()) {
+            for (MicroserviceInstance instance : cseServices) {
+                List<String> eps = instance.getEndpoints();
+                for (String ep : eps) {
+                    URIEndpointObject uri = new URIEndpointObject(ep);
+                    instances.add(new DefaultServiceInstance(instance.getServiceId(), uri.getHostOrIp(),
+                            uri.getPort(), false));
+                }
             }
         }
-        }
         return instances;
     }
 
     @Override
     public ServiceInstance getLocalServiceInstance() {
-        /*ServiceRegistryClient client = RegistryClientFactory.getRegistryClient();
-        String appId = DynamicPropertyFactory.getInstance().getStringProperty("APPLICATION_ID", "default").get();
-        String versionRule = DynamicPropertyFactory.getInstance().getStringProperty("service_description.version", "1.0.0").get();
-        String cseServiceID = client.getMicroserviceId(appId , "springmvc", versionRule);
-        List<MicroserviceInstance> cseServices = client.getMicroserviceInstance(cseServiceID, cseServiceID);
-        DefaultServiceInstance serviceInstance = null;
-        if(null != cseServices && !cseServices.isEmpty())
-        {
-        for(MicroserviceInstance instance:cseServices) 
-        {
-            List<String> eps=instance.getEndpoints();
-            for(String ep:eps)
-            {
-                URIEndpointObject uri = new URIEndpointObject(ep);
-                serviceInstance =  new DefaultServiceInstance("springmvc", uri.getHostOrIp(), uri.getPort(), false);
-            }
-        }
-        }
-        return serviceInstance;*/
         return null;
```

Comment:
exception will be printed/logged as error by zuul router , as its deprecated null check is handled